I believe the changes in trace broke a test:
```
FAIL: fs_test.go:37: S.TestFetchesFilesystem

fs_test.go:74:
    c.Assert(err, ErrorMatches, testCase.err, comment)
... error string = "" +
...     "failed to determine filesystem type on /dev/foo: error\n" +
...     "\terror"
... regex string = "error, failed to determine filesystem type on /dev/foo: error"
... lsblk fails

OOPS: 0 passed, 1 FAILED
--- FAIL: TestSystem (0.01s)
FAIL
FAIL	github.com/gravitational/gravity/lib/system	0.144s
```

--- a/lib/rpc/deploy.go
+++ b/lib/rpc/deploy.go
@@ -92,7 +92,7 @@ func (r *DeployAgentsRequest) CheckAndSetDefaults() error {
 		}
 	}
 	if r.Progress == nil {
-		r.Progress = utils.NewConsoleProgress(context.TODO(), "", 0)
+		r.Progress = utils.NewNopProgress()
 	}
 	return nil
 }

--- a/lib/system/fs_test.go
+++ b/lib/system/fs_test.go
@@ -63,7 +63,7 @@ xfs
 		},
 		{
 			lsblk:   failingRunner{trace.Errorf("error")},
-			err:     `error, failed to determine filesystem type on /dev/foo: error`,
+			err:     "failed to determine filesystem type on /dev/foo: error\n\terror",
 			comment: "lsblk fails",
 		},
 	}

--- a/tool/gravity/cli/gc.go
+++ b/tool/gravity/cli/gc.go
@@ -175,7 +175,7 @@ func newCollector(env *localenv.LocalEnvironment) (*vacuum.Collector, error) {
 		clusterEnv:   clusterEnv,
 		proxy:        proxy,
 	}
-	creds, err := deployAgents(ctx, req)
+	creds, err := deployAgents(ctx, env, req)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}

--- a/tool/gravity/cli/rpcagent.go
+++ b/tool/gravity/cli/rpcagent.go
@@ -293,6 +293,7 @@ func newDeployAgentsRequest(ctx context.Context, req deployAgentsRequest) (*rpc.
 		LeaderParams:   req.leaderParams,
 		Leader:         req.leader,
 		NodeParams:     req.nodeParams,
+		Progress:       utils.NewConsoleProgress(ctx, "", 0),
 	}, nil
 }
 

--- a/tool/gravity/cli/rpcagent.go
+++ b/tool/gravity/cli/rpcagent.go
@@ -187,7 +187,7 @@ func rpcAgentDeployHelper(ctx context.Context, localEnv *localenv.LocalEnvironme
 
 	localCtx, cancel := context.WithTimeout(ctx, defaults.AgentDeployTimeout)
 	defer cancel()
-	return deployAgents(localCtx, req)
+	return deployAgents(localCtx, localEnv, req)
 }
 
 func verifyCluster(ctx context.Context,
@@ -246,8 +246,8 @@ func upsertRPCCredentialsPackage(
 	return secretsPackage, nil
 }
 
-func deployAgents(ctx context.Context, req deployAgentsRequest) (credentials.TransportCredentials, error) {
-	deployReq, err := newDeployAgentsRequest(ctx, req)
+func deployAgents(ctx context.Context, env *localenv.LocalEnvironment, req deployAgentsRequest) (credentials.TransportCredentials, error) {
+	deployReq, err := newDeployAgentsRequest(ctx, env, req)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
@@ -266,7 +266,7 @@ func deployAgents(ctx context.Context, req deployAgentsRequest) (credentials.Tra
 }
 
 // newDeployAgentsRequest creates a new request to deploy agents on the local cluster
-func newDeployAgentsRequest(ctx context.Context, req deployAgentsRequest) (*rpc.DeployAgentsRequest, error) {
+func newDeployAgentsRequest(ctx context.Context, env *localenv.LocalEnvironment, req deployAgentsRequest) (*rpc.DeployAgentsRequest, error) {
 	servers, err := verifyCluster(ctx, req.clusterState, req.proxy)
 	if err != nil {
 		return nil, trace.Wrap(err)
@@ -293,7 +293,7 @@ func newDeployAgentsRequest(ctx context.Context, req deployAgentsRequest) (*rpc.
 		LeaderParams:   req.leaderParams,
 		Leader:         req.leader,
 		NodeParams:     req.nodeParams,
-		Progress:       utils.NewConsoleProgress(ctx, "", 0),
+		Progress:       utils.NewProgress(ctx, "", 0, bool(env.Silent)),
 	}, nil
 }
 

--- a/tool/gravity/cli/update.go
+++ b/tool/gravity/cli/update.go
@@ -124,7 +124,7 @@ func newUpdater(ctx context.Context, localEnv, updateEnv *localenv.LocalEnvironm
 	defer cancel()
 	logger.WithField("request", req).Debug("Deploying agents on cluster nodes.")
 	localEnv.PrintStep("Deploying agents on cluster nodes")
-	creds, err := deployAgents(deployCtx, req)
+	creds, err := deployAgents(deployCtx, localEnv, req)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}

